src/main/java/com/kmbl/InventoryManagementService/models/Inventory.java

```diff
@@ -7,30 +7,51 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
-
 import java.util.List;
 
 @Data
 @NoArgsConstructor
 @DynamoDBTable(tableName = "inventory")
 public class Inventory {
 
+    @DynamoDBAttribute
+    List<Integer> pincodes;
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
     @DynamoDBAutoGeneratedKey
-    @DynamoDBHashKey
+    @DynamoDBHashKey(attributeName = "id")
```

Comment:
no need to add attribute name, it is self understood, it is only added when colomun name is different from field name